flash/image/classification/data.py

```diff
@@ -380,3 +444,236 @@ def show_post_tensor_transform(self, samples: List[Any], running_stage: RunningS
+            train_data: The input to :meth:`~flash.core.data.data_source.DataSource.load_data` to use when creating
+                the train dataset.
+            val_data: The input to :meth:`~flash.core.data.data_source.DataSource.load_data` to use when creating
+                the validation dataset.
+            test_data: The input to :meth:`~flash.core.data.data_source.DataSource.load_data` to use when creating
+                the test dataset.
+            predict_data: The input to :meth:`~flash.core.data.data_source.DataSource.load_data` to use when creating
+                the predict dataset.
+            train_transform: The dictionary of transforms to use during training which maps
+                :class:`~flash.core.data.process.Preprocess` hook names to callable transforms.
+            val_transform: The dictionary of transforms to use during validation which maps
+                :class:`~flash.core.data.process.Preprocess` hook names to callable transforms.
+            test_transform: The dictionary of transforms to use during testing which maps
+                :class:`~flash.core.data.process.Preprocess` hook names to callable transforms.
+            predict_transform: The dictionary of transforms to use during predicting which maps
+                :class:`~flash.core.data.process.Preprocess` hook names to callable transforms.
+            data_fetcher: The :class:`~flash.core.data.callback.BaseDataFetcher` to pass to the
+                :class:`~flash.core.data.data_module.DataModule`.
+            preprocess: The :class:`~flash.core.data.data.Preprocess` to pass to the
+                :class:`~flash.core.data.data_module.DataModule`. If ``None``, ``cls.preprocess_cls`` will be
+                constructed and used.
+            val_split: The ``val_split`` argument to pass to the :class:`~flash.core.data.data_module.DataModule`.
+            batch_size: The ``batch_size`` argument to pass to the :class:`~flash.core.data.data_module.DataModule`.
+            num_workers: The ``num_workers`` argument to pass to the :class:`~flash.core.data.data_module.DataModule`.
+            sampler: The ``sampler`` to use for the ``train_dataloader``.
+            preprocess_kwargs: Additional keyword arguments to use when constructing the preprocess. Will only be used
+                if ``preprocess = None``.
+
+        Returns:
+            The constructed data module.
+
+        Examples::
+
+            data_module = DataModule.from_data_source(
+                DefaultDataSources.FOLDERS,
+                train_data="train_folder",
+                train_transform={
+                    "to_tensor_transform": torch.as_tensor,
+                },
+            )
+        """
+
+        preprocess = preprocess or cls.preprocess_cls(
+            train_transform,
+            val_transform,
+            test_transform,
+            predict_transform,
+            **preprocess_kwargs,
+        )
```

Comment:
Again, passing in transforms as raw dictionaries results in a confusing interface for this function and unwieldy code. Now the data module has to know about how to make a preprocessor out of transforms. While handy, it adds complexity to the interface that doesn't need to be there.